index.js

```diff
@@ -1,196 +1,64 @@
-    return browserifyBundler.bundle()
-      .on('error', function(err, b) {
-        delete err.stream;
-        log.error('[BROWSERIFY] @ ' + currentDateTime());
-        log.warn(err.toString());
-        return true;
-      })
-      .pipe(source('app.js'))
-      .pipe(buffer())
-      .pipe($.util.env.type !== 'production' ? $.sourcemaps.init({loadMaps: true}) : $.util.noop())
-      .pipe($.util.env.type === 'production' ? $.uglify() : $.util.noop())
-      .pipe($.util.env.type !== 'production' ? $.sourcemaps.write('./') : $.util.noop())
-      .pipe(gulp.dest(paths.dist + '/' + paths.scripts));
-  };
-  gulp.task('build-app', buildApp);
+  /* Returns public configuration API */
+  return {
+    /**
+     * @method transform
+     * @param transform {String or Object} either a string referencing the transform to perform or an object
+     *                  can either be a string identifying the transform or an object of the format:
+     *                    {name: 'name of transform', options: {// transform options}}
+     */
+    transform: function(transform) {
+      bundleOptions.browserify.transforms.push(transform);
 
-  /* Full build */
-  gulp.task('build', function(callback) {
-    browserifyBundler = browserify(watchify.args);
-    configureBrowserify();
-    runSequence('clean',
-      [
-        'copy-resources',
-        'build-styles',
-        'build-app'
-      ], function() {
-        log.mark('[BROWSERIFY] complete!');
-        callback();
+      bundleOptions = mergeBundleDefaults({
+        browserify: {
+          transforms: options.browserify.transforms
+        }
       });
-  });
-
-  /* Watch build */
-  gulp.task('watch', function() {
-    browserifyBundler = watchify(browserify(watchify.args));
-    configureBrowserify();
-    gulp.watch(paths.app + '/' + paths.styles + '/**/*.scss', buildStyles);
-    gulp.watch([
-      paths.app + '/**/*.*',
-      '!' + paths.app + '/**/*.+(' + nonResources +')',
-    ], copyResource);
-    browserifyBundler.on('update', buildApp);
-    browserifyBundler.on('log', function(data) {
-      log.mark('[BROWSERIFY] ' + data.toString());
-    });
-    runSequence('clean',
-      [
-        'copy-resources',
-        'build-styles',
-        'build-app'
-      ]);
-  });
 
-  /* Returns public configuration API */
-  return {
-    transform: function(fn) {
-      transformStack.push(fn);
+      buildTask.options(bundleOptions);
+      watchTask.options(bundleOptions);
     },
 
-    nonResources: function(nrsc) {
-      nonResources = nrsc;
-    },
+    /**
+     * @method nonResources
+     * @param excludes {String} regex that defines which files to exclude from the copy-resources task.
+     */
+    nonResources: function(excludes) {
+      bundleOptions = mergeBundleDefaults({
+        browserify: {
+          excludes: excludes
+        }
+      });
 
-    configurePaths: function(options) {
-      _.extend(paths, options);
+      buildTask.options(bundleOptions);
```

Comment:
I need to change this api to a pure callthrough to buildTask.nonResources() and watchTask.nonResources(), same with the rest of the api (options() and transform()).